packages/core/src/components/overlay/_overlay.scss

```diff
@@ -95,3 +90,35 @@ $overlay-background-color: rgba($black, 0.7);
     outline: none;
   }
 }
+
+// we surreptitiously add this class to the overlay content container so that
+// users won't need to explicitly manage the position mode for inline and
+// non-inline rendering.
+.pt-overlay-content {
+  position: fixed;
+}
+
+// when rendered inline, we want the overlay to position itself relative to
+// its parent container, not relative to the whole window. thus, we change
+// to position:absolute.
+.pt-overlay-inline {
+  .pt-overlay-backdrop,
+  .pt-overlay-content {
+    position: absolute;
+  }
+}
+
+// scroll container for non-inline overlay. it needs an explicit z-index to
+// ensure that the backdrop renders on top of any navbars.
+.pt-overlay-scroll-container.pt-overlay-open {
```

Comment:
i wonder if we still need this class? but that's another issue, about scrolling dialogs etc